### DISCUSSION TITLE
an open source project using go language

### DISCUSSION THREAD

**Discussion by abeloa3411** [SpaceyaTech/support#9] [General]
Hello, i was asking if there is anybody here who knows go language so that we ca also have an open source project using go language

**Comment by musaubrian** [comment]
That would be great.
If you happen to get one, I would love to be a part of it

**Reply by abeloa3411** [reply]
sorry for the late reply i have an idea, how will i share it with you

**Reply by musaubrian** [reply]
No worries.
Here is good, or over on discord

**Reply by oyamo** [reply]
Did you settle on a project?

**Reply by musaubrian** [reply]
Not yet,
You got one?

**Reply by abeloa3411** [reply]
Ok i found an article where a developer created  a production crud api with golang, mongodb, nginx and docker swarm cluster.... i feel like it's great way to learn about api's that are in production and arranged proffessionally....we can base our project on something like that......an Api that has all the technolgy needed to be in a production environment

**Reply by oyamo** [reply]
Like a real world project?

**Reply by musaubrian** [reply]
Like something that SpaceYaTech could use or...
What about the git-metrics project they're building in java?

**Reply by musaubrian** [reply]
Could you share the article

**Reply by abeloa3411** [reply]
https://it-wars.com/posts/virtualisation/docker-exemple-golang-mongodb-rest-api/

**Comment by abeloa3411** [comment]
Yes

**Reply by JohnKinyanjui** [reply]
Hello there i know golang

**Reply by abeloa3411** [reply]
You can join if you are interested

**Reply by aibunny** [reply]
i'm not good in go but I would like to join

**Comment by oyamo** [comment]
How about a secret manager 4 securely storing credentials (Keys/Certs/Secrets) in a vault.


Server
- Manage secrets
- GitHub hooks for downloading secrets for GitHub actions
- Monitor secrets

CLI
- Upload secrets
- Download Secrets/Keys/Certs

**Reply by musaubrian** [reply]
So kinda like the github CLI's secrets command

**Reply by abeloa3411** [reply]
Should we go with this project?

**Reply by musaubrian** [reply]
yes it's great

**Comment by oyamo** [comment]
Like harshicorp vault

**Reply by oyamo** [reply]
This @musaubrian  https://www.hashicorp.com/products/vault 
Just simple secret management operations

**Reply by musaubrian** [reply]
aah, got it

**Comment by brf153** [comment]
Hey I also want to be a part of the golang project. Also, I have an idea for the project

**Comment by brf153** [comment]
We can actually build a Vercel Clone using golang and next.js

**Comment by abeloa3411** [comment]
So between the two project, the secret vault and the vercel clone, what are we agreeing on.

**Reply by oyamo** [reply]
Okay. I'll defend the **secret vault project**.  I think we will get a chance to learn advanced things like
 - building Github actions hooks
 - cryptography & Sec
 - High availability and scalability 
 - compliance and Auditing
 
 Furthermore, not that Vercel is a bad Idea, it just requires javascript knowledge for building, bundling/compiling Javascript application just like Vercel does

**Reply by abeloa3411** [reply]
Ok lets work on the secret vault project.....So who can lead us through this project

**Reply by brf153** [reply]
Yeah I would also like to contribute to the secret vault

**Reply by brf153** [reply]
But please explain what exactly needs to be done in it

**Reply by oyamo** [reply]
We need to build a go server and client(CLI) for it. We can plan for it. 
But we don't have a Golang channel on discord

**Reply by musaubrian** [reply]
made a request for the channel to be created, you can check if its created.

**Reply by brf153** [reply]
I think we should make a vault but rather than deploying on vercel like platforms we can try to implement what these platform actually do( which is to first build and run the commands and then deploy).. I think we will need kubernetes for it...Personally, I wanted to understand how does vercel build the projects and deploys it. If someone has idea, please share

**Comment by musaubrian** [comment]
what about something like [sendit](https://sendit.sh/)
Although it can be a feature in the vault project

**Comment by brf153** [comment]
So, let us start working on the project

**Comment by brf153** [comment]
first we will be needing a plan

**Comment by oyamo** [comment]
Hi @abeloa3411 @brf153 @musaubrian  @JohnKinyanjui and anyone interested in the project. Kindly fill your name and email on this form so we can plan
FORM:  https://forms.office.com/r/fjCY3rMMZe

**Reply by JimmyTron** [reply]
https://discord.gg/5kdxTSfQ
Join this server make it your Home.

**Comment by JimmyTron** [comment]
Join the GO-LANG DEVS SERVER HERE
https://discord.gg/5kdxTSfQ

**Reply by oyamo** [reply]
Thank you Jimmy

**Reply by abeloa3411** [reply]
joined....think that is the best platform for communication
